nqvinh/array/find_missing_number/main.c

```diff
@@ -0,0 +1,91 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+
+/*
+1. How do you find the missing number in a given integer array of 1 to 100?
+SOLUTIONS:
+	Sumary of full array minus to Sumary of missing number array.
+*/
+
+/*
+	Build: gcc main.c -o main
+	Run:   ./main 55
+*/
+
+void mixArray(unsigned int array[], unsigned int arraySize)
+{
+	int i, tmp;
+	for (i = 0; i < arraySize; i++)
+	{
+		// (rand() % (upper - lower + 1)) + lower;
+		// upper = 99, lower = 0;
+		int a = (rand() % (99 - 0 + 1)) + 0;
+		int b = (rand() % (99 - 0 + 1)) + 0;
+
+		// swap 2 array member
+		tmp      = array[a];
+		array[a] = array[b];
+		array[b] = tmp;
+	}
+}
+
+void initArray(unsigned int array[], unsigned int missingNum, unsigned int arraySize)
+{
+	if ((missingNum < 1) || (missingNum > 100))
+	{
+		printf("Wrong missing number\n");
+		exit(0);
+	}
+
+	for (int i = 0; i < arraySize; i++)
+	{
+		if ((i+1) == missingNum)
+		{
+			array[i] = 0;
+			continue;
+		}
+		else
+		{
+			array[i] = i + 1;
+		}
+	}
+
+	mixArray(array, arraySize);
+}
+
+unsigned int findMissingNumber(unsigned int array[])
+{
+	unsigned int orgSum = 0;
+	unsigned int curSum = 0;
+
+	// sum from 5 to 88: (5+88)*84:2 = 3906
+	orgSum = ((1+100)*100)/2;
+
+	for (int i = 0; i < 100; i++)
```

Comment:
As above comment, the array size is now only 99

Reply:
fixed